src/sysext/sysext.c

```diff
@@ -262,9 +277,38 @@ static int unmerge_hierarchy(
                 if (r == 0)
                         break;
 
+                r = read_one_line_file(work_dir_info_file, &base64_work_dir);
+                if (r < 0) {
+                        if (r != -ENOENT)
+                                return log_error_errno(r, "Failed to read '%s': %m", work_dir_info_file);
+                }
+
+                if (base64_work_dir) {
+                        r = unbase64mem(base64_work_dir, (void**)&work_dir, NULL);
```

Comment:
why base64 this btw? a path in the fs should be safe to write there without need for encoding this in base64

Reply:
The path could have a newline in it and I wanted to have a single line file. It's a fringe case, so maybe I'm paranoid.

Reply:
hmm, i see. maybe use xescape() or cescape()? keeps stuff readablle at least